src/service/xhr-impl.js

```diff
@@ -451,37 +466,72 @@ export function assertSuccess(response) {
   });
 }
 
-
 /**
- * Response object in the Fetch API.
+ * Response object in the Fetch API. This interface is implemented by native
+ * `Response` and by the polyfill and mock response types.
  *
  * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+ *
+ * @record
```

Comment:
What I was trying to do here was create a structural interface that would be satisfied by all three kinds of response object: browser-native `Response`, `FetchPolyfillResponse` as defined in this file, and `MockFetchResponse` as defined in `xhr-mock.js`. I'm not sure it entirely worked, since I still had to put type assertions in a couple places. How would you advise doing this?